examples/hybrid_jobs/6_QNSPSA_optimizer_with_embedded_simulator/qnspsa_with_embedded_simulator.ipynb

```diff
@@ -0,0 +1,1360 @@
+{
```

Comment:
<p>The original paper [1] seems to have the loss converge to analytical target. Is it because QAOA is harder than the examples in the original paper, so we do not converge well?</p><p> </p><p> </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/aws/amazon-braket-examples/pull/149/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='37'/>

Reply:
Not sure if my understanding is correct: From the "Concentration" discussion in the QAOA paper (https://arxiv.org/pdf/1411.4028.pdf), we know that QAOA performance is not expected to be too good with a limited p. Basically when p value is low, the space from the ansatz might not cover some of the better solutions. It is then not about the optimizer's performance.

On the other hand, QAOA ansatz is a more complicated one comparing to the toy model in the original paper, and I noticed that initial config matters. I saw GD and QNG stuck to local minimum without further optimization over the loss function quite often.  This seems to be not so much a problem with QN-SPSA. Will share more results on that.